packages/lesswrong/components/posts/PostsPage/PostsPage.tsx

```diff
@@ -670,22 +704,28 @@ const PostsPage = ({fullPost, postPreload, eagerPostComments, refetch, classes}:
     {showRecommendations && recommendationsPosition === "right" && fullPost && <PostSideRecommendations post={fullPost} />}
   </>;
 
-  // check for deep equality between terms and eagerPostComments.terms
-  const useEagerResults = eagerPostComments && isEqual(commentTerms, eagerPostComments?.terms);
-
-  const lazyResults = useMulti({
-    terms: {...commentTerms, postId: post._id},
-    skip: useEagerResults,
-    ...postsCommentsThreadMultiOptions,
-  });
-
-
   // If this is a non-AF post being viewed on AF, redirect to LW.
   if (isAF && !post.af) {
     const lwURL = "https://www.lesswrong.com" + location.url;
     return <PermanentRedirect url={lwURL}/>
   }
 
+  // Redirect from ?commentId=... to #... if we should always show comments in context
+  if (queryCommentId && commentPermalinkStyleSetting.get() === 'in-context') {
+    const urlSearchParams = new URLSearchParams(location.query);
+    urlSearchParams.delete('commentId');
+    const otherQueryParams = urlSearchParams.toString();
+    const hashVersion = `#${queryCommentId}`;
+    const redirectUrl = `${location.pathname}${otherQueryParams ? `?${otherQueryParams}` : ''}${hashVersion}`;
+
+    if (isServer) {
+      return <PermanentRedirect url={redirectUrl} status={302} />
+    } else {
+      // navigate doesn't result in the whole page unmounting momentarily
+      navigate(redirectUrl, { replace: true });
+    }
```

Comment:
1. `PermanentRedirect` isn't actually permanent, it uses the status given
2. This `PermanentRedirect`/`navigate` distinction is required to fix the issue of scrolling to the top and back down, but I'll need to change `ScrollToTop` as well to fully fix it

Reply:
Perhaps this should live in `PostsPageWrapper` instead of here?

Reply:
I think it makes sense here because:
- It uses `queryCommentId`, which is also used by other stuff in this file
- There is another `PermanentRedirect` in this file already

Reply:
Ok, I'm not sure how much this matters, but there's a good chunk of code that runs before the redirects, so I'm inclined to move them up to the wrapper component (nbd either way though)